src/htm/regions/SPRegion.cpp

```diff
@@ -349,7 +349,7 @@ Spec *SPRegion::createSpec() {
                     1,                                // elementCount
                     "",                               // constraints
                     "0.1",                            // defaultValue
-                    ParameterSpec::ReadOnlyAccess)); // access
+                    ParameterSpec::ReadWriteAccess)); // access
```

Comment:
Please don't. Sorry it isn't really clear from the current code, it is a WIP in #561 , `synPermConnected` is essentially const, since it's used in Connections during initialization (and cannot, by its logic, be changed after that). A code that would change that variable would break all previous learning. 

You mentioned this was "another obstacle" for you, how is that?

Reply:
I need to be able to set this in creating my SPRegion or is there another way of doing this?

Reply:
I guess I'll meanwhile rollback this change and get your help to deal with this issue.

Reply:
> I need to be able to set this in creating my SPRegion or is there another way of doing this?

You can create a SPRegion, just don't set the RO variables. 
If you wanted to change the synPermConnected after initialization of the region, don't. The previous learning will be lost, and the code shouldn't allow it.

Reply:
No. I want to set the synPermConnected at SPRegion creation. This is normally done by means of the parameters dict. However, synPermConnected has ReadOnlyAccess and hence that entry in the parameter dict is rejected.

Reply:
Maybe the bug is that the initialization parameter being rejected.

Reply:
> I want to set the synPermConnected at SPRegion creation. This is normally done by means of the parameters dict. However, synPermConnected has ReadOnlyAccess and hence that entry in the parameter dict is rejected.

FYI @dkeeney , I think this is a limitation of the NetworkAPI (?). We want to be able to set some variables during initializer-time (equivalent of `SP(..., synPermConnected=0.222)` but the var. should be read-only after that during runtime (so `sp.setSynPermConnected(xxx)` is not allowed). 

@fcr you could workaround the inability to change synPermConnected by appropriately changing `synPermActiveInc/Dec`, it has the same effect.

Reply:
>FYI @dkeeney , I think this is a limitation of the NetworkAPI (?). We want to be able to set some variables during initializer-time (equivalent of SP(..., synPermConnected=0.222) but the var. should be read-only after that during runtime (so sp.setSynPermConnected(xxx) is not allowed).

Exactly. 
This is a serous bug IMHO.
Because of this limitation, I had to set the access for all the parameters in the advanced/regions to ReadWrite in order to allow initialization of them, even though they should not be altered afterwards.

Reply:
The Spec should reflect what the code implements.  So just changing the spec would allow the setParameterXXX to be called but it would probably not be handled by SPRegion unless you change the setParameterXXX( ) function as well to handle it.  But if the algorithm cannot accept it as @breznak mentioned then you would probably break something.

By the way.  A parameter passed in during creation should have an access of CreateAccess rather than ReadOnlyAccess.  CreateAccess means can be used for initialization but read-only after that.  ReadOnlyAccess implies that it is something created by the algorithm and is readable.

Reply:
>@fcr you could workaround the inability to change synPermConnected by appropriately changing synPermActiveInc/Dec, it has the same effect.

Same effect but wrong semantics.

Reply:
> By the way. A parameter passed in during creation should have an access of CreateAccess rather than ReadOnlyAccess. CreateAccess means can be used for initialization but read-only after that. ReadOnlyAccess implies that it is something created by the algorithm and is readable.

Great! 
Can you please change synPermConnected to CreateAccess?
I'm not touching the C++ stuff any more. Don't want to break anything. :-)

Reply:
Once this PR is finally accepted, I'll go through and start applying the CreateAccess to the relevant region specs and make a new (smaller) PR.

Reply:
BTW never came across CreatAccess before.

Reply:
tracked in #669